cli_tools/gce_vm_image_import/importer/inflater.go

```diff
@@ -146,3 +294,146 @@ func getDisk(workflow *daisy.Workflow, diskIndex int) *daisy.Disk {
 
 	panic("Did not find CreateDisks step.")
 }
+
+func (inflater apiInflater) traceLogs() []string {
+	return inflater.serialLogs
+}
+
+func (inflater apiInflater) addTraceLog(l string) {
+	inflater.serialLogs = append(inflater.serialLogs, l)
+}
+
+func (inflater apiInflater) inflate(ctx context.Context) (persistentDisk, error) {
+	startTime := time.Now()
+	diskName := fmt.Sprintf("shadow-disk-%v", inflater.args.ExecutionID)
+	cd := computeAlpha.Disk{
+		Name:                diskName,
+		SourceStorageObject: inflater.args.SourceFile,
+	}
+	err := inflater.computeClient.CreateDiskAlpha(inflater.args.Project, inflater.args.Zone, &cd)
+	if err != nil {
+		return persistentDisk{}, err
+	}
+	defer func() {
+		// Cleanup the shadow disk ignoring error
+		inflater.computeClient.DeleteDisk(inflater.args.Project, inflater.args.Zone, cd.Name)
+	}()
+
+	bkt, objPath, err := storage.GetGCSObjectPathElements(inflater.args.SourceFile)
+	if err != nil {
+		return persistentDisk{}, err
+	}
+	sourceFile := inflater.storageClient.GetObject(bkt, objPath).GetObjectHandle()
+	attrs, err := sourceFile.Attrs(ctx)
+	if err != nil {
+		return persistentDisk{}, daisy.Errf("Failed to get source file attributes: %v", err)
+	}
+	sourceFileSizeGb := (attrs.Size - 1) / 1073741824 + 1
+
+	diskUri := fmt.Sprintf("zones/%s/disks/%s", inflater.args.Zone, diskName)
+	pd := persistentDisk{
+		uri:           diskUri,
+		sizeGb:        cd.SizeGb,
+		sourceGb:      sourceFileSizeGb,
+		sourceType:    "vmdk", // only vmdk is supported right now
+		inflationTime: time.Since(startTime),
+	}
+
+	inflater.addTraceLog("Started checksum calculation.")
+	pd.checksum, err = inflater.calculateChecksum(ctx, diskUri)
+	return pd, err
+}
+
+// run a workflow to calculate checksum, and cleanup the disk
+func (inflater apiInflater) calculateChecksum(ctx context.Context, diskUri string) (string, error) {
```

Comment:
The 'cleanup disk' comment is confusing since the caller has a `defer` to delete the disk.

Reply:
Deleted.